Sources/Fullscreen/Explore/ExploreView.swift

```diff
@@ -177,8 +245,19 @@ public final class ExploreView: UIView {
 
 extension ExploreView: UICollectionViewDelegate {
     public func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
-        let item = sections[indexPath.section].items[indexPath.item]
-        delegate?.exploreView(self, didSelectItem: item, at: indexPath)
+        if indexPath.section < exploreSections.count {
+            let item = exploreSections[indexPath.section].items[indexPath.item]
+            delegate?.exploreView(self, didSelectItem: item, at: indexPath)
+        } else {
+            let item = recommendationsSection[indexPath.row]
+            delegate?.exploreViewRecommendations(self, didSelectItemAtIndex: indexPath.row, withId: item.id)
+        }
+    }
+
+    public func collectionView(_ collectionView: UICollectionView, willDisplay cell: UICollectionViewCell, forItemAt indexPath: IndexPath) {
+        if indexPath.section == exploreSections.count {
+            delegate?.exploreViewRecommendations(self, willDisplayItemAtIndex: indexPath.row)
```

Comment:
This seems like a misleading name for a method, since it'll only be called for each item in the last section. How about either:
- Calling it for all items, and changing the `Int` to `IndexPath`
- Changing the method signature to:

```swift
func exploreViewRecommendations(_ adRecommendationsGridView: ExploreView, willDisplayItemForLastSectionAtIndex index: Int)
```